Sources/MongoSwift/ReadConcern.swift

```diff
@@ -14,85 +14,104 @@ public class ReadConcern: Codable {
+       return level == nil
+    }
+
+    // Initializes a new `ReadConcern` with the same level as the provided `mongoc_read_concern_t`.
+    // The caller is responsible for freeing the original `mongoc_read_concern_t`.
+    internal init(from readConcern: OpaquePointer) {
+        if let level = mongoc_read_concern_get_level(readConcern) {
+            self.level = Level(rawValue: String(cString: level))
+        }
     }
 
-    /// Initialize a new `ReadConcern` from a `ReadConcern.Level`.
-    public convenience init(_ level: Level) {
-        self.init(level.rawValue)
+    /// Initializes a new `ReadConcern` by copying an existing `ReadConcern`.
+    public init(from readConcern: ReadConcern) {
+        self.level = readConcern.level
+    }
+
+    /// Initialize a new `ReadConcern` with a `Level`.
+    public init(_ level: Level) {
+        self.level = level
     }
 
     /// Initialize a new `ReadConcern` from a `String` corresponding to a read concern level.
     public init(_ level: String) {
-        self._readConcern = mongoc_read_concern_new()
-        mongoc_read_concern_set_level(self._readConcern, level)
+        self.level = Level(rawValue: level)
     }
 
     /// Initialize a new empty `ReadConcern`.
     public init() {
-        self._readConcern = mongoc_read_concern_new()
+        self.level = nil
     }
 
     /// Initializes a new `ReadConcern` from a `Document`.
-    public convenience init(_ doc: Document) {
+    public init(_ doc: Document) {
         if let level = doc["level"] as? String {
             self.init(level)
         } else {
             self.init()
         }
     }
 
-    /// Initializes a new `ReadConcern` by copying an existing `ReadConcern`.
-    public init(from readConcern: ReadConcern) {
-        self._readConcern = mongoc_read_concern_copy(readConcern._readConcern)
-    }
-
-    /// Initializes a new `ReadConcern` by copying a `mongoc_read_concern_t`.
-    /// The caller is responsible for freeing the original `mongoc_read_concern_t`.
-    internal init(from readConcern: OpaquePointer?) {
-        self._readConcern = mongoc_read_concern_copy(readConcern)
-    }
-
     private enum CodingKeys: String, CodingKey {
         case level
     }
 
-    public required convenience init(from decoder: Decoder) throws {
-        let container = try decoder.container(keyedBy: CodingKeys.self)
-        if let level = try container.decodeIfPresent(String.self, forKey: .level) {
-            self.init(level)
-        } else {
-            self.init()
-        }
-    }
-
-    public func encode(to encoder: Encoder) throws {
-        var container = encoder.container(keyedBy: CodingKeys.self)
-        try container.encodeIfPresent(self.level, forKey: .level)
-    }
-
-    /// Cleans up internal state.
-    deinit {
-        guard let readConcern = self._readConcern else {
-            return
+    /**
+     * Creates a new `ReadConcern` with the provided options and passes it to the provided closure.
```

Comment:
I would reword this to say "Creates a new `mongoc_read_concern_t` based on this `ReadConcern` and passes it to the provided closure."

Also "The pointer is only valid within the body of the closure and will be freed after the body completes."

Reply:
This still needs to be updated.